README.md

```diff
@@ -7,23 +7,27 @@ This repository provides a [Terraform module](https://learn.hashicorp.com/tutori
-1. The [docker-mirror module](https://registry.terraform.io/modules/sourcegraph/executors/aws/4.1.0/submodules/docker-mirror) provisions a Docker registry pull-through cache.
-1. The [networking module](https://registry.terraform.io/modules/sourcegraph/executors/aws/4.1.0/submodules/networking) provisions a network to be shared by the executor and Docker registry resources.
-1. The [credentials module](https://registry.terraform.io/modules/sourcegraph/executors/aws/4.1.0/submodules/credentials) provisions credentials required by the Sourcegraph instance to enable observability and auto-scaling of executors.
+2. The [docker-mirror module](https://registry.terraform.io/modules/sourcegraph/executors/aws/4.1.0/submodules/docker-mirror) provisions a Docker registry pull-through cache.
+3. The [networking module](https://registry.terraform.io/modules/sourcegraph/executors/aws/4.1.0/submodules/networking) provisions a network to be shared by the executor and Docker registry resources.
+4. The [credentials module](https://registry.terraform.io/modules/sourcegraph/executors/aws/4.1.0/submodules/credentials) provisions credentials required by the Sourcegraph instance to enable observability and auto-scaling of executors.
 
 The [multiple-executors example](https://github.com/sourcegraph/terraform-aws-executors/blob/v4.1.0/examples/multiple-executors) uses the submodule directly to provision multiple executor resource groups performing different types of work. Follow this example if you are:
 
 1. Provisioning executors for use with multiple features (e.g., both [auto-indexing](https://docs.sourcegraph.com/code_intelligence/explanations/auto_indexing) and [server-side batch changes](https://docs.sourcegraph.com/batch_changes/explanations/server_side)), or
-1. Provisioning resources for multiple Sourcegraph instances (e.g., test, prod)
+2. Provisioning resources for multiple Sourcegraph instances (e.g., test, prod)
 
 This repository also provides a [root module](https://registry.terraform.io/modules/sourcegraph/executors/aws/4.1.0) combining the executors, network, and docker-mirror resources into an easier to use package.
 
 The [single-executor example](https://github.com/sourcegraph/terraform-aws-executors/blob/v4.1.0/examples/single-executor) uses the root module to provision a single executor type. Follow this example if you are deploying to a single Sourcegraph instance and using a single executors-backed feature.
 
 ## Requirements
 
-- [Terraform](https://www.terraform.io/) ~> 1.1.0
-- [hashicorp/aws](https://registry.terraform.io/providers/hashicorp/aws/3.0.0) ~> 3.0.0
+- [Terraform](https://www.terraform.io/) 
+  - 4.1.0 and below: `~> 1.1.0`
+  - 4.2.0 and above: `>= 1.1.0, < 2.0.0`
+- [hashicorp/aws](https://registry.terraform.io/providers/hashicorp/aws) 
+  - 4.1.0 and below: `~> 3.0.0`
+  - 4.2.0 and above: `>= 3.0, ~> 4.0`
```

Comment:
Isn't this the same? Asking, because I find this easier to read than `~>` and that's also the notation we use in l27

```suggestion
  - 4.2.0 and above: `>= 3.0, < 5.0.0`
```